src/java.base/share/classes/javax/crypto/JceSecurityManager.java

```diff
@@ -100,11 +103,15 @@ CryptoPermission getCryptoPermission(String alg) {
         // javax.crypto.* packages.
         // NOTE: javax.crypto.* package maybe subject to package
         // insertion, so need to check its classloader as well.
-        Class<?>[] context = getClassContext();
+        final List<StackFrame> stack = AccessController.doPrivileged(
+                (PrivilegedAction<List<StackFrame>>)
+                        () -> StackWalker.getInstance(
+                                Option.RETAIN_CLASS_REFERENCE)
+                                .walk((s) -> s.collect(Collectors.toList())));
+
```

Comment:
Note: StackWalker is a stateless capability object. It's not the walk() method that requires a permission, but the creation of the StackWalker itself (hence my suggestion to create it in the constructor, or in a static initializer). If you walk the stack from with a doPrivileged call then the doPrivileged frame will appear in the returned `List<StackFrame>`; this may (or may not) be OK - depending on the logic that processes the stack.

Reply:
Thanks.  I was going to step through this code more thoroughly today, hopefully I would have caught that.  

This code is only needed in certain deployment and Cipher creation situations, so would rather not create a static CodeWalker that is not normally used.

Reply:
Fair enough.